rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/CharacteristicLongWriteOperation.java

```diff
@@ -209,4 +214,58 @@ public Boolean call(Object emission) {
             }
         };
     }
+
+    private static Func1<Observable<? extends Throwable>, Observable<?>> retryOperationStrategy(
+            final WriteOperationRetryStrategy writeOperationRetryStrategy,
+            final ByteBuffer byteBuffer,
+            final int batchSize) {
+        return new Func1<Observable<? extends Throwable>, Observable<?>>() {
+            @Override
+            public Observable<?> call(Observable<? extends Throwable> emittedOnWriteFailure) {
+                return emittedOnWriteFailure
+                        .flatMap(canRetryError())
+                        .map(toLongWriteFailureObject())
+                        .compose(writeOperationRetryStrategy)
+                        .doOnNext(replaceByteBufferPositionForRetry());
+            }
+
+            @NonNull
+            private Func1<Throwable, WriteOperationRetryStrategy.LongWriteFailure> toLongWriteFailureObject() {
+                return new Func1<Throwable, WriteOperationRetryStrategy.LongWriteFailure>() {
+                    @Override
+                    public WriteOperationRetryStrategy.LongWriteFailure call(Throwable throwable) {
+                        final int failedBatchNumber = (byteBuffer.position() - batchSize) / batchSize;
```

Comment:
This calculation will be wrong if the last batch will not be equal size to default `batchSize`. Ideally we should be passing the batch size to this function from the writeBatchAndObserve().

Reply:
So `writeBatchAndObserve()` should emit a DTO instead of a `ByteAssociation<UUID>` like:
```
class WriteBatch {
    final ByteAssociation<UUID> byteAssociation;
    final int batchSize;
}
```

Reply:
Indeed, but what advantage compared to something like that instead:
```
if (byteBuffer.hasRemaining()) {
    return (byteBuffer.position() - batchSize) / batchSize;
} else {
    return byteBuffer.position() / batchSize;
}
```

Reply:
Ideally we should emit the batches to write `byte[]` which should be indexed. Those pairs of bytes and indexes should then be written (and retried) accordingly to the strategy to be in line with in as immutable approach as possible. I am not sure how would the performance turn up...